index.js

```diff
@@ -64,30 +64,37 @@ exports.indexLinks = function (message, opts, each) {
     opts = { rel: opts }
   if (!opts)
     opts = {}
-  var msg  = opts.msg
+  var msg = opts.msg
   var feed = opts.feed
   var blob = opts.blob
-  var any  = !(msg || feed || blob)
+  var any = !(msg || feed || blob)
 
   traverse(toMsgContent(message), function (obj, rel) {
     if (opts.rel && rel !== opts.rel) return
 
-    var r = (typeof obj == 'string') ? obj : obj.link
+    var r;
+    if (typeof obj == 'string') {
+      r = obj
+    } else if (typeof obj.link !== "undefined") {
```

Comment:
Is there any case where this wouldn't be a string? We might be able to be more specific here.

Reply:
@christianbundy as far as I know, a mentions array can be something like:

```
[
  "#patchfox",
  {link: "#patchfox"},
  {name: "andre", link: "48634786"}
]
```
So you can have a literal string, or an object with a link but you shouldn't have something that is not one of those two cases but since we're talking about people building their own clients, they will make mistakes, there will be broken messages and we might want to code around it because one it is on the feed, it is there kinda forever.

Reply:
Right, I agree with you. My concern is:

```json
[
  "#patchfox",
  {
    "link": { "foo": { "bar": [] } }
  }
]
```

This passes `typeof obj.link !== 'undefined'`, but if we take that value and pass it to ssb-ref and friends as a *string* then it seems like we'll break something. You know what I mean?

Reply:
oooohhh, I see... makes sense. Let me amend the commit

Reply:
Pushed a change with a better check, it is down below.